### PR TITLE
Add a SLURM example with minimal config

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

This PR adds a new SLURM example specifically for when users want to specify an `accelerate` config yaml in addition to regular launching. This is *specifically* useful as generally SLURM resources are dynamically created/set in SLURM, such as:
```
export LAUNCHER="accelerate launch \
    --num_processes $((SLURM_NNODES * GPUS_PER_NODE)) \
    --num_machines $SLURM_NNODES \
    --rdzv_backend c10d \
    --main_process_ip $head_node_ip \
    --main_process_port 29500 \
    "
```
However since we only care about these values, this PR shows how you can also pass in a config yaml (if it's stored somewhere locally/on a shared FS):

```
export LAUNCHER="accelerate launch \
    --config ${ACCELERATE_DIR}/examples/slurm/fsdp_config.yaml \
    --num_processes $((SLURM_NNODES * GPUS_PER_NODE)) \
    --num_machines $SLURM_NNODES \
    --rdzv_backend c10d \
    --main_process_ip $head_node_ip \
    --main_process_port 29500 \
    "
```
With that config file being *extremely* minimal:
```
distributed_type: FSDP
fsdp_config:
  fsdp_activation_checkpointing: false
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
  fsdp_backward_prefetch: BACKWARD_PRE
  fsdp_cpu_ram_efficient_loading: true
  fsdp_forward_prefetch: false
  fsdp_offload_params: false
  fsdp_sharding_strategy: FULL_SHARD
  fsdp_state_dict_type: SHARDED_STATE_DICT
  fsdp_sync_module_states: true
  fsdp_use_orig_params: true
```

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@SunMarc

### CODE REVIEW

File: examples/slurm/submit_multinode_fsdp.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#### Set network #####
+######################
+head_node_ip=$(scontrol show hostnames $SLURM_JOB_NODELIST | head -n 1)
+######################
+export ACCELERATE_DIR="${ACCELERATE_DIR:-/accelerate}"
+
+export LAUNCHER="accelerate launch \
+    --config ${ACCELERATE_DIR}/examples/slurm/fsdp_config.yaml \
+    --num_processes $((SLURM_NNODES * GPUS_PER_NODE)) \
+    --num_machines $SLURM_NNODES \
```
Comment (SunMarc) [inline examples/slurm/submit_multinode_fsdp.sh:30]: $SLURM_NNODES is not set

File: examples/slurm/submit_multinode_fsdp.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    --main_process_port 29500 \
+    "
+export SCRIPT="${ACCELERATE_DIR}/examples/complete_nlp_example.py"
+export SCRIPT_ARGS=" \
+    --mixed_precision fp16 \
+    --output_dir ${ACCELERATE_DIR}/examples/output \
+    "
+    
+# This step is necessary because accelerate launch does not handle multiline arguments properly
+export CMD="$LAUNCHER $PYTHON_FILE $ARGS"
```
Comment (SunMarc) [inline examples/slurm/submit_multinode_fsdp.sh:42]: forgot to change these
```suggestion
export CMD="$LAUNCHER $SCRIPT $SCRIPT_ARGS" 
```

**Review by SunMarc — APPROVED** [review]
Thanks for adding this ! Maybe in a future PR, we can add a section in accelerate on what this script does line by line !

File: examples/slurm/submit_multinode_fsdp.sh (hunk 1 above)
Comment (muellerzr) [reply examples/slurm/submit_multinode_fsdp.sh:30]: That just "exists" in a SLURM node :)

Comment (SunMarc) [reply examples/slurm/submit_multinode_fsdp.sh:30]: Slurm magic 🪄😂 Sounds good !

**Pull request merged**